### PR TITLE
kurtosis udaf

### PR DESCRIPTION
**Pull request by jatin510**
## Which issue does this PR close?

Closes #12250 https://github.com/apache/datafusion/issues/12250

## Rationale for this change

I followed the changes in PR: https://github.com/apache/datafusion/pull/12273

## What changes are included in this PR?

Added a udaf to calculate the kurtosis.

## Are these changes tested?

Added the test in main.rs file took reference from `mode` udaf.

### CODE REVIEW

**Comment by s5dsn-eqee** [comment]
@jatin510 Thank you 
Could you add information about the function to the README.

File: src/lib.rs [hunk 1]
```diff
@@ -26,14 +26,15 @@ use datafusion::logical_expr::AggregateUDF;
 #[macro_use]
 pub mod macros;
 pub mod common;
+pub mod kurtosis;
 pub mod mode;
 
 pub mod expr_extra_fn {
     pub use super::mode::mode;
```
Comment (s5dsn-eqee) [inline src/lib.rs]: pls add
`pub use super::kurtosis::kurtosis;`

Comment (jatin510) [reply src/lib.rs]: done

File: README.md [hunk 2]
```diff
@@ -77,10 +77,20 @@ SELECT min_by(x, y) FROM VALUES (1, 10), (2, 5), (3, 15), (4, 8) as tab(x, y);
...
+- +-------------------+
+
 ```
 
 ## Done
 
 - [x] `mode(expression) -> scalar` - Returns the most frequent (mode) value from a column of data.
 - [x] `max_by(expression1, expression2) -> scalar` - Returns the value of `expression1` associated with the maximum value of `expression2`.
 - [x] `min_by(expression1, expression2) -> scalar` - Returns the value of `expression1` associated with the minimum value of `expression2`.
+- [x] `kurtosis(expression) -> scalar` - Returns the kurtosis, a measure of the `tailedness` of the distribution, for the given numeric values in `expression`.
```
Comment (dharanad) [inline README.md]: This documentation is not same as one in code, we should have consistent documentation everywhere.

File: src/kurtosis.rs [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    signature: Signature,
+}
+
+impl Debug for KurtosisFunction {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("KurtosisFunction")
+            .field("signature", &self.signature)
+            .finish()
+    }
+}
```
Comment (dharanad) [inline src/kurtosis.rs:50]: I think `#[derive(Debug)]` macro should serve the purpose here.

**Review by dharanad — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -77,10 +77,20 @@ SELECT min_by(x, y) FROM VALUES (1, 10), (2, 5), (3, 15), (4, 8) as tab(x, y);
 -- | 2                   |
 -- +---------------------+
 
+-- Get the kurtosis value of a column of numeric data.
```
Comment (s5dsn-eqee) [inline README.md]: It seems we have enough example, just a description in "Done" section, will be fine.

File: src/kurtosis.rs (hunk 3 above)
Comment (s5dsn-eqee) [reply src/kurtosis.rs:50]: It seems that an explicit implementation of Debug will be more informative here.

File: README.md (hunk 4 above)
Comment (jatin510) [reply README.md]: made all the mentioned changes @dmitrybugakov

File: README.md (hunk 2 above)
Comment (jatin510) [reply README.md]: made all the mentioned changes @dharanad

**Review by dharanad — APPROVED** [review]
LGTM. Thanks @jatin510

**Review by s5dsn-eqee — APPROVED** [review]
lgtm 
@jatin510 thank you
@dharanad thanks for the help

**Pull request merged**